src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -2300,8 +2300,13 @@ public String getContextPath() throws IOException {
          * Use {@link #createCrumbedUrl} instead if you intend to call {@link WebRequest#setRequestBody}, typical of a POST request.
          */
         public WebRequest addCrumb(WebRequest req) {
-            NameValuePair crumb = getCrumbHeaderNVP();
-            req.setRequestParameters(Arrays.asList(crumb));
+            ArrayList<NameValuePair> params = new ArrayList<>();
+            params.add(getCrumbHeaderNVP());
+            List<NameValuePair> oldParams = req.getRequestParameters();
+            if (oldParams != null && !oldParams.isEmpty()) {
```

Comment:
Can `getRequestParameters` actually be null? I doubt it. And if it is empty, the `addAll` is a no-op anyway; so probably the `if` statement could be elided.

Reply:
they can be if you call `setRequestParameters` with `null` (its not checked - but they are initialised to an empty collection)